src/main/java/com/ncov/module/service/MaterialRequiredService.java

```diff
@@ -0,0 +1,107 @@
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * 物料寻求服务
+ * @author lucas
+ */
+@Slf4j
+@Service
+public class MaterialRequiredService extends ServiceImpl<MaterialRequiredMapper, MaterialRequiredEntity> {
+
+    @Autowired
+    private MaterialRequiredMapper materialRequiredMapper;
+    @Autowired
+    private UserContext userContext;
+
+    /**
+     * 根据相关条件，查询物料寻求分页列表
+     * @return
+     */
+    public com.ncov.module.controller.resp.Page<MaterialResponse> getRequiredPageList(int pageSize, int pageNums, String category){
+        IPage<MaterialRequiredEntity> result = materialRequiredMapper.selectPage(
+                new Page<MaterialRequiredEntity>()
+                        .setPages(pageSize)
+                        .setSize(pageNums),
+                new LambdaQueryWrapper<MaterialRequiredEntity>()
+                        .like(MaterialRequiredEntity::getMaterialSuppliedCategory, category));
+        com.ncov.module.controller.resp.Page<MaterialResponse> page = new com.ncov.module.controller.resp.Page<>();
+        page.setData(this.carry(result.getRecords()));
+        page.setPage(result.getPages());
+        page.setTotal(result.getTotal());
+        page.setPageSize(result.getSize());
+        return page;
+    }
+
+    /**
+     * 保存物料寻求信息
+     * @param materialRequest
+     * @return
+     */
+    @Transactional(rollbackFor = Exception.class)
+    public MaterialRequiredEntity saveRequiredInfo(MaterialRequest materialRequest){
+        MaterialRequiredEntity materialRequiredEntity = MaterialRequiredEntity.builder()
+                .materialRequiredComment(materialRequest.getComment())
+                .materialRequiredContactorName(materialRequest.getContactorName())
+                .materialRequiredContactorPhone(materialRequest.getContactorPhone())
+                .materialRequiredOrganizationId(1L)
+                .materialRequiredUserId(1L)
+                .materialSuppliedOrganizationName(materialRequest.getOrganisationName())
+                .materialRequireStatus("PUBLISHED")
+                .materialRequiredReceivedAddress(materialRequest.getAddress())
+                .materialSuppliedImageUrls(Joiner.on(",").join(materialRequest.getImageUrls()))
+                .gmtCreated(new Date()).build();
+        materialRequest.getMaterials().stream().forEach(item->{
+            materialRequiredEntity.setMaterialSuppliedCategory(item.getCategory());
+            materialRequiredEntity.setMaterialSuppliedName(item.getName());
+            materialRequiredEntity.setMaterialSuppliedStandard(item.getStandard());
+            materialRequiredEntity.setMaterialRequiredQuantity(item.getQuantity());
+            materialRequiredMapper.insert(materialRequiredEntity);
```

Comment:
Just for discussion, will `saveBatch` be easier compared with this foreach insert?